app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarScreen.kt

```diff
@@ -422,21 +428,46 @@ fun FilterBox(
     }
 }
 
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
 fun Week(
     modifier: Modifier = Modifier,
     tasks: List<Task>
 ) {
-    Row(
-        modifier = modifier
-            .horizontalScroll(rememberScrollState())
-            .fillMaxWidth()
-    ) {
-        (1..7).forEach {
-            Day(
-                name = DayOfWeek.of(it).name,
-                tasks = tasks.filter { task -> task.scheduledAt.dayOfWeek.value == it }
-            )
+    val days = (1..7).map { DayOfWeek.of(it).name }
+    val pagerState = rememberPagerState(pageCount = { days.size })
+    val currentPage = pagerState.currentPage
+
+    @Composable
+    fun DayPage(day: String, pageIndex: Int) {
```

Comment:
Would be easier to read if these components would be exported out of the parent component. If these components are not supposed to be called from outside, you should also make them `private`.